### ISSUE TITLE
单卡24G内存，多卡如何部署qwen-image-edit

### ISSUE THREAD

**Issue by hh1562535601** [QwenLM/Qwen-Image#86]
单卡24G内存，多卡如何部署qwen-image-edit

**Comment by yangsheng0522** [comment]
貌似部署不了

**Comment by Iriya99** [comment]
https://github.com/deepbeepmeep/Wan2GP

**Comment by wings0820** [comment]
gguf-int3 VL int4 + Qwen-Image-Lightning 4步，20gb显存、在3090上20s内出图，效果暂时没有看出和原版明显的区别

**Comment by lumingkai234** [comment]
可以请教楼上具体如何做的吗   如果想把qwen-image-edit  部署在3090上是否可行

**Comment by Tumb1eweed** [comment]
> gguf-int3 VL int4 + Qwen-Image-Lightning 4步，20gb显存、在3090上20s内出图，效果暂时没有看出和原版明显的区别

求教 这个是怎么做到的 我用单卡24G加nunchaku版本的qwen image edit每次出图都是一张纯黑图…

**Comment by WeeeNHao** [comment]
> > gguf-int3 VL int4 + Qwen-Image-Lightning 4步，20gb显存、在3090上20s内出图，效果暂时没有看出和原版明显的区别
> 
> 求教 这个是怎么做到的 我用单卡24G加nunchaku版本的qwen image edit每次出图都是一张纯黑图…

是不是单精度运行的，应该是因为数值下溢了，你尝试了什么其他方法嘛？

**Comment by Tumb1eweed** [comment]
> > > gguf-int3 VL int4 + Qwen-Image-Lightning 4步，20gb显存、在3090上20s内出图，效果暂时没有看出和原版明显的区别
> > 
> > 
> > 求教 这个是怎么做到的 我用单卡24G加nunchaku版本的qwen image edit每次出图都是一张纯黑图…
> 
> 是不是单精度运行的，应该是因为数值下溢了，你尝试了什么其他方法嘛？

有点搞…我把transformers版本升级了下就好了

**Comment by WeeeNHao** [comment]
> > > > gguf-int3 VL int4 + Qwen-Image-Lightning 4步，20gb显存、在3090上20s内出图，效果暂时没有看出和原版明显的区别
> > > 
> > > 
> > > 求教 这个是怎么做到的 我用单卡24G加nunchaku版本的qwen image edit每次出图都是一张纯黑图…
> > 
> > 
> > 是不是单精度运行的，应该是因为数值下溢了，你尝试了什么其他方法嘛？
> 
> 有点搞…我把transformers版本升级了下就好了

请问老哥用的哪个版本，我是transfomers4.56.1，diffusers0.35.2，还是有这个问题

**Comment by Tumb1eweed** [comment]
> > > > > gguf-int3 VL int4 + Qwen-Image-Lightning 4步，20gb显存、在3090上20s内出图，效果暂时没有看出和原版明显的区别
> > > > 
> > > > 
> > > > 求教 这个是怎么做到的 我用单卡24G加nunchaku版本的qwen image edit每次出图都是一张纯黑图…
> > > 
> > > 
> > > 是不是单精度运行的，应该是因为数值下溢了，你尝试了什么其他方法嘛？
> > 
> > 
> > 有点搞…我把transformers版本升级了下就好了
> 
> 请问老哥用的哪个版本，我是transfomers4.56.1，diffusers0.35.2，还是有这个问题
transformers==4.56.2
diffusers==0.35.1

我把代码也贴出来

import math
import torch
from diffusers import FlowMatchEulerDiscreteScheduler, QwenImageEditPipeline
from nunchaku import NunchakuQwenImageTransformer2DModel
scheduler_config = {
    "base_image_seq_len": 256,
    "base_shift": math.log(3),  # We use shift=3 in distillation
    "invert_sigmas": False,
    "max_image_seq_len": 8192,
    "max_shift": math.log(3),  # We use shift=3 in distillation
    "num_train_timesteps": 1000,
    "shift": 1.0,
    "shift_terminal": None,  # set shift_terminal to None
    "stochastic_sampling": False,
    "time_shift_type": "exponential",
    "use_beta_sigmas": False,
    "use_dynamic_shifting": True,
    "use_exponential_sigmas": False,
    "use_karras_sigmas": False,
}
scheduler = FlowMatchEulerDiscreteScheduler.from_config(scheduler_config)

# Load the model
transformer = NunchakuQwenImageTransformer2DModel.from_pretrained("/hy-tmp/nunchaku-qwen-image-edit/svdq-int4_r128-qwen-image-edit-lightningv1.0-8steps.safetensors")

pipeline = QwenImageEditPipeline.from_pretrained(
    "/hy-tmp/Qwen-Image-Edit", transformer=transformer, scheduler=scheduler, torch_dtype=torch.bfloat16
)


transformer.set_offload(
    True, use_pin_memory=False, num_blocks_on_gpu=8
)  # increase num_blocks_on_gpu if you have more VRAM
pipeline._exclude_from_cpu_offload.append("transformer")
pipeline.enable_sequential_cpu_offload()
